### PR TITLE
Enable almost all tc39 tests

### PR DESCRIPTION
**Pull request by MStoykov**

### CODE REVIEW

**Comment by MStoykov** [comment]
Please go commit by commit ;)

File: js/tc39/tc39_test.go [hunk 1]
```diff
@@ -495,16 +489,19 @@ func (ctx *tc39TestCtx) runTC39Tests(name string) {
...
 		newName := path.Join(name, file.Name())
-		if pathBasedBlock[newName] {
-			ctx.t.Run(newName, func(t *testing.T) {
-				t.Skipf("Skip %s beause of path based block", newName)
-			})
-			continue
+		for _, path := range pathBasedBlock { // TODO: use trie / binary search?
+			if strings.HasPrefix(newName, path) {
+				ctx.t.Run(newName, func(t *testing.T) {
+					t.Skipf("Skip %s beause of path based block", newName)
```
Comment (imiric) [inline js/tc39/tc39_test.go:501]:
```suggestion
					t.Skipf("Skipped %s because of path based block", newName)
```

**Review by imiric — DISMISSED** [review]
`breaking_test_errors.json` is impossible to review :sweat_smile:, but LGTM otherwise.

Running all of it in ~15m doesn't sound bad considering the scope. Nice work! :clap:

**Review by na-- — DISMISSED** [review]
LGTM, though I haven't reviewed all of the new errors in any detail... Hopefully we bring them down with further goja updates :crossed_fingers:

**Review by na-- — APPROVED** [review]
LGTM, though please squash when merging - the fewer copies of the huge JSON file we have in the git history, the better...

**Comment by MStoykov** [comment]
>  the fewer copies of the huge JSON file we have in the git history, 

git only tracks diffs ... so this is barely a problem, especially given the size of the `vendor` directory

**Review by na-- — APPROVED** [review]

**Pull request merged**